tests/steamship_tests/client/test_client.py

```diff
@@ -16,3 +19,56 @@ def test_user():
     user = User.current(client).data
     assert user.id is not None
     assert user.handle is not None
+
+
+def test_client_creation_with_new_autogen_space():
+    client1 = get_steamship_client(create_space=True)
+    client2 = get_steamship_client()
+    client3 = get_steamship_client()
+
+    assert client2.config.space_id is None
+    assert client2.config.space_handle is None
+    assert client3.config.space_id is None
+    assert client3.config.space_handle is None
+
+    assert client1.config.space_id is not None
+    assert client1.config.space_handle is not None
+
+    Space(client=client1, id=client1.config.space_id).delete()
+
+
+def test_client_creation_space_failure_modes():
+    # Can't create a space by specifying an ID
+    with pytest.raises(SteamshipError):
+        get_steamship_client(space_id="foo", create_space=True)
+
+    client1 = get_steamship_client(create_space=True)
+
+    # Can't create a space with a handle that already exists
+    with pytest.raises(SteamshipError):
+        get_steamship_client(space_handle=client1.config.space_handle, create_space=True)
```

Comment:
In the future, I would like the client to be in charge of deciding which space to use. I worked with 2 different spaces for my app, and passing the space handle to all the calls was dreadful.

Reply:
What do you think about removing the `space_id` and `space_handle` arguments from all our top-level APIs? It seems to me they aren't really necessary anyway, so just add API noise.

The `Configuration` object within the client already provides a way to grab a client-handle that's hard-pinned to a particular space, just as you mention preferring above (I prefer that too)